src/components/signinPage/Input.tsx

```diff
@@ -45,64 +47,51 @@ const ImgPosition = styled.div`
   right: 18px;
 `;
 
-const ErrorMesage = styled.p`
-  color: #ff5b56;
-  font-size: 14px;
-  font-weight: 400;
-`;
-
 interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
   id: string;
   placeholder: string;
-  error?: boolean;
-  label: string;
+  errors: FieldErrors;
+  label?: string;
+  validation?: UseFormRegisterReturn;
 }
 
-function Input({ id, type = "text", placeholder, error, label }: InputProps) {
+function Input({
+  id,
+  type = "text",
+  placeholder,
+  errors,
+  label,
+  validation,
+}: InputProps) {
   const [pwState, setPwState] = useState(false);
-  const passwordRef = useRef<HTMLInputElement>(null);
-
-  const {
-    register,
-    formState: { errors },
-  } = useForm();
+  const error = errors[id];
 
   const toggleEyesButton = () => {
-    if (passwordRef.current) {
-      if (pwState) {
-        passwordRef.current.type = "text";
-        setPwState(false);
-      } else {
-        passwordRef.current.type = "password";
-        setPwState(true);
-      }
-    }
+    setPwState((prev) => !prev);
   };
 
   return (
     <>
       <Layout>
         <Label htmlFor={id}>{label}</Label>
         <InputForm
-          type={type}
-          {...register(id, { required: true })}
+          id={id}
+          type={pwState ? "password" : "text"}
           placeholder={placeholder}
-          ref={passwordRef}
-          $error={error}
+          $error={!!error}
+          {...validation}
         />
         {type === "password" && (
           <ImgPosition onClick={toggleEyesButton}>
             {pwState ? (
-              <Image src={setPwOff} alt="Pw-off" />
-            ) : (
               <Image src={setPwOn} alt="Pw-on" />
+            ) : (
+              <Image src={setPwOff} alt="Pw-off" />
             )}
           </ImgPosition>
         )}
       </Layout>
-      {errors.category?.type === "required" && (
-        <ErrorMesage>필수 입력사항입니다.</ErrorMesage>
-      )}
+      {errors && <ErrorMesage text={error?.message?.toString()} />}
```

Comment:
r:
- `errors`가 아니라 `error`가 되어야 할 것 같아요.
- 앞에서 error가 [truthy](https://developer.mozilla.org/ko/docs/Glossary/Truthy)로 검증되었기 때문에 error에서 message 호출할 때는 체이닝하지 않아도 될 것 같습니다.

```suggestion
      {errors && <ErrorMesage text={error.message?.toString()} />}
```